### PR TITLE
improve pretty printing

### PR DESCRIPTION
**Pull request by c-cube**
cf #140

### CODE REVIEW

File: lib/pretty.ml [hunk 1]
```diff
@@ -3,7 +3,7 @@ let pp_list sep ppx out l =
   let pp_sep out () = Format.fprintf out "%s@ " sep in
   Format.pp_print_list ~pp_sep ppx out l
 
-let rec format std (out:Format.formatter) (x:t) : unit =
+let rec format inbox std (out:Format.formatter) (x:t) : unit =
```
Comment (Leonidas-from-XIV) [inline lib/pretty.ml:6]: Can you change this to `in_box`? I can't get my mind to not read it as "inbox".

File: CHANGES.md [hunk 2]
```diff
@@ -41,6 +41,7 @@
 - Avoid copying unnecessarily large amounts of strings when parsing (#85, #108,
   @Leonidas-from-XIV)
 - Fix `stream_to_file` (#133, @tcoopman and @gasche)
+- Fix pretty-printing behavior for lists/objects inside lists
```
Comment (Leonidas-from-XIV) [inline CHANGES.md:44]: This should actually go into a new `unreleased` section, I'll push this on `master`.

**Review by Leonidas-from-XIV — CHANGES_REQUESTED** [review]
Already much better not to have so many dangling opening parens. I'll fix the changelog on `master`.

File: CHANGES.md [hunk 3]
```diff
@@ -8,6 +8,8 @@
 
 ### Fixed
 
+- Fix pretty-printing behavior for lists/objects inside lists
```
Comment (Leonidas-from-XIV) [inline CHANGES.md:11]:
```suggestion
- Fix pretty-printing behavior for lists/objects inside lists (#141, @c-cube)
```

**Comment by mjambon** [comment]
Don't you want to reserve wrapping (hov box) for only sequences of "atoms" (null, boolean, number, string, `{}`, `[]`) like it was done before, and print the rest either horizontally or vertically (hv box)? This requires inspecting the list of items to determine the formatting style but it's lightweight. Please let me know if a PR would help.

**Comment by c-cube** [comment]
Ah, that's interesting. I didn't realize it was done this way, but it makes sense (hv box for big objects is nicer). I think it can wait but that's just my personal opinion, and if you want to make a PR it'll make @Leonidas-from-XIV happy :)

**Comment by mjambon** [comment]
@c-cube ok, the PR is in! #142

**Pull request merged**